test/metabase/lib/drill_thru/automatic_insights_test.cljc

```diff
@@ -0,0 +1,260 @@
+
+(deftest ^:parallel do-not-return-fk-filter-for-non-fk-column-test
+  (testing "underlying-records should only get shown once for aggregated query (#34439)"
+    (let [test-case           {:click-type  :cell
+                               :query-type  :aggregated
+                               :column-name "max"}
+          {:keys [query row]} (lib.drill-thru.tu/query-and-row-for-test-case test-case)
+          context             (lib.drill-thru.tu/test-case-context query row test-case)]
+      (testing (str "\nQuery = \n"   (u/pprint-to-str query)
+                    "\nContext =\n" (u/pprint-to-str context))
+        (let [drills (lib/available-drill-thrus query context)]
+          (testing (str "\nAvailable drills =\n" (u/pprint-to-str drills))
+            (is (= 1
+                   (count (filter #(= (:type %) :drill-thru/underlying-records)
+                                  drills))))))))))
+
+
+#?(:cljs (comment metabase.test-runner.assert-exprs.approximately-equal/keep-me))
+
+(def last-month
+  #?(:cljs (let [now    (js/Date.)
+                 year   (.getFullYear now)
+                 month  (.getMonth now)]
+             (-> (js/Date.UTC year (dec month))
+                 (js/Date.)
+                 (.toISOString)))
+     :clj  (let [last-month (-> (jt/zoned-date-time (jt/year) (jt/month))
+                                (jt/minus (jt/months 1)))]
+             (jt/format :iso-offset-date-time last-month))))
+
+(defn- underlying-state [query agg-index agg-value breakout-values exp-filters-fn]
+  (let [columns                         (lib/returned-columns query)
+        {aggs      :source/aggregations
+         breakouts :source/breakouts}   (group-by :lib/source columns)
+        agg-column                      (nth aggs agg-index)
+        agg-dim                         {:column     agg-column
+                                         :column-ref (lib/ref agg-column)
+                                         :value      agg-value}]
+    (is (= (count breakouts)
+           (count breakout-values)))
+    (let [breakout-dims (for [[breakout value] (map vector breakouts breakout-values)]
+                          {:column     breakout
+                           :column-ref (lib/ref breakout)
+                           :value      value})
+          context       (merge agg-dim
+                               {:row        (cons agg-dim breakout-dims)
+                                :dimensions breakout-dims})]
+      (is (=? {:lib/type :mbql/query
+               :stages [{:filters     (exp-filters-fn agg-dim breakout-dims)
+                         :aggregation (symbol "nil #_\"key is not present.\"")
+                         :breakout    (symbol "nil #_\"key is not present.\"")
+                         :fields      (symbol "nil #_\"key is not present.\"")}]}
+              (->> (lib.drill-thru/available-drill-thrus query context)
+                   (m/find-first #(= (:type %) :drill-thru/underlying-records))
```

Comment:
All these tests look like they were copypasted from the underlying records test namespace, they're even testing for underlying records drills and not automatic-insights

Reply:
They were copy-pasted and I didn't finish porting the file. It's done now.